extensions/resteasy-reactive/quarkus-resteasy-reactive/runtime/src/main/java/io/quarkus/resteasy/reactive/server/runtime/security/EagerSecurityHandler.java

```diff
@@ -66,18 +68,36 @@ public void handle(ResteasyReactiveRequestContext requestContext) throws Excepti
         if (!authorizationController.isAuthorizationEnabled()) {
             return;
         }
+
+        // if proactive auth is disabled, then accessing SecurityIdentity is a blocking operation
+        // setting identity here will enable SecurityInterceptors registered in Quarkus Security Deployment to run checks
+        final boolean setSecurityIdentity;
+        if (requestContext instanceof QuarkusResteasyReactiveRequestContext) {
+            setSecurityIdentity = ((QuarkusResteasyReactiveRequestContext) requestContext).isProactiveAuthenticationDisabled;
+        } else {
+            setSecurityIdentity = false;
+        }
+
         requestContext.requireCDIRequestScope();
         SecurityCheck theCheck = check;
         if (!theCheck.isPermitAll()) {
             requestContext.suspend();
-            getCurrentIdentityAssociation().get().getDeferredIdentity().map(new Function<SecurityIdentity, Object>() {
-                @Override
-                public Object apply(SecurityIdentity securityIdentity) {
-                    theCheck.apply(securityIdentity, methodDescription,
-                            requestContext.getParameters());
-                    return null;
-                }
-            })
+            final var currentIdentityAssociation = getCurrentIdentityAssociation().get();
+            currentIdentityAssociation.getDeferredIdentity()
+                    .call(securityIdentity -> {
+                        if (setSecurityIdentity && securityIdentity != null) {
```

Comment:
This is a very minor comment, I wonder would it make sense to make it a bit more verbose and move the `if (setSecurityIdentity == true)` outside to avoid an extra `Uni.call()` step ? If `setSecurityIdentity == false` -> run the existing Uni sequence otherwise the one this PR introduces ? Feel free to ignore this suggestion if you think it won't save much

Reply:
Good idea, I improved it a little bit. Thanks